float32[2] endPoint1
float32[2] midPoint
float32[2] endPoint2